pyterrier_dr/bge_m3.py

```diff
@@ -0,0 +1,77 @@
+from pyterrier.transformer import Transformer
+from FlagEmbedding import BGEM3FlagModel
```

Comment:
FlagEmbedding will need to be added as a dependency or it will break. I think I'd prefer to make it an optional dependency to avoid installing extra stuff they don't need if somebody isn't using this model.